### PR TITLE
Internal: Update php unit tests with PHP 7.4, 8.0 [ED-12973]

### PR DESCRIPTION
**Pull request by TzviRabinovitch**

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -218,22 +218,22 @@
...
             "support": {
                 "issues": "https://github.com/justinrainbow/json-schema/issues",
-                "source": "https://github.com/justinrainbow/json-schema/tree/5.2.12"
+                "source": "https://github.com/justinrainbow/json-schema/tree/v5.2.13"
             },
-            "time": "2022-04-13T08:02:27+00:00"
+            "time": "2023-09-26T02:20:38+00:00"
         },
         {
             "name": "matipojo/elementor-editor-testing",
```
Comment (davseve) [inline composer.lock:226]: Change the name to `elementor/elementor-editor-testing`

File: tests/phpunit/elementor/app/import-export/test-module.php [hunk 2]
```diff
@@ -12,14 +12,14 @@ class Test_Module extends Elementor_Test_Base {
...
-	public function setUp() {
+	public function setUp():void  {
 		parent::setUp();
 
 		$this->elementor_upload_dir = ( new Server() )->get_system_path( Server::KEY_PATH_ELEMENTOR_UPLOADS_DIR );
 		$this->elementor_upload_dir_permission = fileperms( $this->elementor_upload_dir );
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/app/import-export/test-module.php:22]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/core/admin/menu/test-admin-menu-manager.php [hunk 3]
```diff
@@ -32,7 +32,7 @@ public function setUp() {
 		$this->original_menu = $menu;
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/core/admin/menu/test-admin-menu-manager.php:35]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/core/admin/test-admin-notices.php [hunk 4]
```diff
@@ -45,9 +45,9 @@ public function test_admin_notices() {
 		$result = ob_get_clean();
 
 		// Assert
-		$this->assertRegExp( '/\<h3\>test title\<\/h3\>/', $result );
-		$this->assertRegExp( '/\<p\>test description\<\/p\>/', $result );
-		$this->assertRegExp( '/data-notice_id="test_id"/', $result );
+		$this->assertMatchesRegularExpression( '/\<h3\>test title\<\/h3\>/', $result );
+		$this-> assertMatchesRegularExpression( '/\<p\>test description\<\/p\>/', $result );
+		$this-> assertMatchesRegularExpression( '/data-notice_id="test_id"/', $result );
```
Comment (davseve) [inline tests/phpunit/elementor/core/admin/test-admin-notices.php:50]:
```suggestion
		$this->assertMatchesRegularExpression( '/\<p\>test description\<\/p\>/', $result );
		$this->assertMatchesRegularExpression( '/data-notice_id="test_id"/', $result );
```

File: tests/phpunit/elementor/core/admin/test-admin.php [hunk 5]
```diff
@@ -18,15 +18,15 @@ class Test_Admin extends Elementor_Test_Base {
...
+	public function setUp():void  {
 		global $pagenow;
 
 		parent::setUp();
 
 		$this->pagenow = $pagenow;
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/core/admin/test-admin.php:29]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/core/admin/test-admin.php [hunk 6]
```diff
@@ -116,7 +116,7 @@ public function test_body_statues_classes() {
 		$result = apply_filters( 'admin_body_class', '' );
 
 		// Assert
-		$this->assertRegExp( '/elementor-editor-inactive/', $result );
+		$this-> assertMatchesRegularExpression( '/elementor-editor-inactive/', $result );
```
Comment (davseve) [inline tests/phpunit/elementor/core/admin/test-admin.php:119]:
```suggestion
		$this->assertMatchesRegularExpression( '/elementor-editor-inactive/', $result );
```

File: tests/phpunit/elementor/core/admin/test-admin.php [hunk 7]
```diff
@@ -133,7 +133,7 @@ public function test_body_statues_classes__when_edit_with_elementor() {
 		$result = apply_filters( 'admin_body_class', '' );
 
 		// Assert
-		$this->assertRegExp( '/elementor-editor-active/', $result );
+		$this-> assertMatchesRegularExpression( '/elementor-editor-active/', $result );
```
Comment (davseve) [inline tests/phpunit/elementor/core/admin/test-admin.php:136]:
```suggestion
		$this->assertMatchesRegularExpression( '/elementor-editor-active/', $result );
```

File: tests/phpunit/elementor/core/base/test-document.php [hunk 8]
```diff
@@ -226,7 +226,7 @@ public function test_get_editor_panel_config_ensure_default_route() {
 		wp_set_current_user( $before_user->ID );
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/core/base/test-document.php:229]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/core/common/modules/event-tracker/test-db.php [hunk 9]
```diff
@@ -27,7 +27,7 @@ public function setUp() {
 		];
 	}
 
-	public static function setUpBeforeClass() {
+	public static function setUpBeforeClass() : void {
```
Comment (davseve) [inline tests/phpunit/elementor/core/common/modules/event-tracker/test-db.php:30]:
```suggestion
	public static function setUpBeforeClass():void {
```

File: tests/phpunit/elementor/core/common/modules/event-tracker/test-db.php [hunk 10]
```diff
@@ -37,7 +37,7 @@ public static function setUpBeforeClass() {
 		DB::reset_table();
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/core/common/modules/event-tracker/test-db.php:40]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/core/editor/data/globals/endpoints/base.php [hunk 11]
```diff
@@ -17,7 +17,7 @@ public function setUp() {
 		wp_set_current_user( $this->factory()->get_administrator_user()->ID);
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/core/editor/data/globals/endpoints/base.php:20]:
```suggestion
	public function tearDown():void {
```

File: tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php [hunk 12]
```diff
@@ -16,14 +16,14 @@
...
-	public function setUp() {
+	public function setUp():void  {
 		parent::setUp();
 
 		$this->original_experiment_default_state = Plugin::$instance->experiments
 			->get_features( Editor::EDITOR_V2_EXPERIMENT_NAME )['default'];
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php:26]:
```suggestion
	public function tearDown():void {
```

File: tests/phpunit/elementor/core/files/test-uploads-manager.php [hunk 13]
```diff
@@ -90,7 +90,7 @@ public static function setUpBeforeClass() {
 		file_put_contents( self::$temp_directory . self::$template_json_file_name, $template_json );
 	}
 
-	public static function tearDownAfterClass() {
+	public static function  tearDownAfterClass(): void {
```
Comment (davseve) [inline tests/phpunit/elementor/core/files/test-uploads-manager.php:93]:
```suggestion
	public static function tearDownAfterClass():void {
```

File: tests/phpunit/elementor/core/files/test-uploads-manager.php [hunk 14]
```diff
@@ -75,7 +75,7 @@ class Test_Uploads_Manager extends Elementor_Test_Base {
 		"type" => "page",
 	];
 
-	public static function setUpBeforeClass() {
+	public static function setUpBeforeClass(): void {
```
Comment (davseve) [inline tests/phpunit/elementor/core/files/test-uploads-manager.php:78]:
```suggestion
	public static function setUpBeforeClass():void {
```

File: tests/phpunit/elementor/core/test-documents-manager.php [hunk 15]
```diff
@@ -31,8 +31,8 @@ public function test_ajax_get_document_config__set_id_build_with_elementor() {
 	public function test_get_new_post_url() {
 		$new_post_url =  Plugin::$instance->documents->get_create_new_post_url();
 
-		$this->assertContains( 'edit.php?action=elementor_new_post&post_type=', $new_post_url );
-		$this->assertContains( '_wpnonce=', $new_post_url );
+		$this->assertStringContainsString( 'edit.php?action=elementor_new_post&post_type=', $new_post_url );
+		$this->assertStringContainsString	( '_wpnonce=', $new_post_url );
```
Comment (davseve) [inline tests/phpunit/elementor/core/test-documents-manager.php:35]:
```suggestion
		$this->assertStringContainsString( '_wpnonce=', $new_post_url );
```

File: tests/phpunit/elementor/core/utils/import-export/test-url.php [hunk 16]
```diff
@@ -17,16 +17,15 @@ class Test_Url extends Elementor_Test_Base {
 
 	private $wp_rewrite;
 
-	public function setUp() {
+	public function setUp(): void {
```
Comment (davseve) [inline tests/phpunit/elementor/core/utils/import-export/test-url.php:20]:
```suggestion
	public function setUp():void {
```

File: tests/phpunit/elementor/core/utils/test-plugins-manager.php [hunk 17]
```diff
@@ -49,7 +49,7 @@ public function setUp() {
 
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/core/utils/test-plugins-manager.php:52]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/data/base/test-controller.php [hunk 18]
```diff
@@ -17,13 +17,13 @@ class Test_Controller extends Elementor_Test_Base {
...
 
-	public function setUp() {
+	public function setUp():void  {
 		parent::setUp();
 
 		$this->manager = Manager::instance();
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/data/base/test-controller.php:26]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/data/base/test-endpoint.php [hunk 19]
```diff
@@ -15,13 +15,13 @@ class Test_Endpoint extends Elementor_Test_Base {
...
 
-	public function setUp() {
+	public function setUp():void  {
 		parent::setUp();
 
 		$this->manager = Manager::instance();
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/data/base/test-endpoint.php:24]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/data/test-manager.php [hunk 20]
```diff
@@ -15,14 +15,14 @@ class Test_Manager extends Elementor_Test_Base {
...
-	public function setUp() {
+	public function setUp():void  {
 		parent::setUp();
 
 		$this->manager = Manager::instance();
 		$this->manager->kill_server();
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/data/test-manager.php:25]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/data/v2/base/data-test-base.php [hunk 21]
```diff
@@ -10,13 +10,13 @@ abstract class Data_Test_Base extends Elementor_Test_Base {
...
 
-	public function setUp() {
+	public function setUp():void  {
 		parent::setUp();
 
 		$this->set_manager();
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (davseve) [inline tests/phpunit/elementor/data/v2/base/data-test-base.php:19]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/includes/base/test-widget-base.php [hunk 22]
```diff
@@ -67,7 +67,7 @@ public function test_render_content__should_render_static() {
 
 		$widget->render_content();
 
-		$this->assertRegExp( '/render_static/', ob_get_clean() );
+		$this-> assertMatchesRegularExpression( '/render_static/', ob_get_clean() );
```
Comment (davseve) [inline tests/phpunit/elementor/includes/base/test-widget-base.php:70]:
```suggestion
		$this->assertMatchesRegularExpression( '/render_static/', ob_get_clean() );
```

File: tests/phpunit/elementor/includes/template-library/test-manager-general.php [hunk 23]
```diff
@@ -15,7 +15,7 @@ class Elementor_Test_Manager_General extends Elementor_Test_Base {
 	protected static $manager;
 	private $fake_post_id = '123';
 
-	public static function setUpBeforeClass() {
+	public static function  setUpBeforeClass(): void {
```
Comment (davseve) [inline tests/phpunit/elementor/includes/template-library/test-manager-general.php:18]:
```suggestion
	public static function setUpBeforeClass(): void {
```

**Review by davseve — COMMENTED** [review]
1. All the `tearDown`, `tearDownAfterClass` and `setUpBeforeClass` classes has double spacing before the function name
2. Many of the `$this-> assertMatchesRegularExpression` has a redundant space before the assertion call.
3. the function type is written in various ways` : void`,` :void`,`: void`, I don't think matters how we do it but it must be consistent. 
Sometimes there is a double spacing after the func type decleration.

File: tests/phpunit/elementor/includes/template-library/test-manager-local.php [hunk 24]
```diff
@@ -13,11 +13,11 @@ class Elementor_Test_Manager_Local extends Elementor_Test_Base {
 	protected static $manager;
 	private $fake_post_id = '123';
 
-	public static function setUpBeforeClass() {
+	public static function  setUpBeforeClass(): void {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/includes/template-library/test-manager-local.php:16]:
```suggestion
	public static function setUpBeforeClass(): void {
```

File: tests/phpunit/elementor/includes/template-library/test-manager-remote.php [hunk 25]
```diff
@@ -16,7 +16,7 @@ class Elementor_Test_Manager_Remote extends Elementor_Test_Base {
 
 
 
-	public static function setUpBeforeClass() {
+	public static function  setUpBeforeClass(): void {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/includes/template-library/test-manager-remote.php:19]: public static function setUpBeforeClass():void {

File: tests/phpunit/elementor/includes/test-utils.php [hunk 26]
```diff
@@ -9,7 +9,7 @@ class Elementor_Test_Utils extends Elementor_Test_Base {
 
 	const BASE_LINK = 'https://elementor.com/pro/?utm_source=wp-role-manager&utm_campaign=gopro&utm_medium=wp-dash';
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/includes/test-utils.php:12]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/modules/announcements/test-module.php [hunk 27]
```diff
@@ -54,7 +54,7 @@ public function setUp() {
 		$this->module = new Module();
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/announcements/test-module.php:57]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/modules/compatibility-tag/test-compatibility-tag-report.php [hunk 28]
```diff
@@ -36,8 +36,8 @@ public function test_get_report_data__with_html_format() {
 
 		// Assert
 		$this->assertArrayHasKey( 'value', $result );
-		$this->assertRegExp( '/<tr><td> A <\/td><td> Compatible <\/td><\/tr>/', $result['value'] );
-		$this->assertRegExp( '/<tr><td> B <\/td><td> Incompatible <\/td><\/tr>/', $result['value'] );
+		$this-> assertMatchesRegularExpression( '/<tr><td> A <\/td><td> Compatible <\/td><\/tr>/', $result['value'] );
+		$this-> assertMatchesRegularExpression( '/<tr><td> B <\/td><td> Incompatible <\/td><\/tr>/', $result['value'] );
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/compatibility-tag/test-compatibility-tag-report.php:40]:
```suggestion
		$this->assertMatchesRegularExpression( '/<tr><td> A <\/td><td> Compatible <\/td><\/tr>/', $result['value'] );
		$this->assertMatchesRegularExpression( '/<tr><td> B <\/td><td> Incompatible <\/td><\/tr>/', $result['value'] );```

File: tests/phpunit/elementor/modules/compatibility-tag/test-module.php [hunk 29]
```diff
@@ -65,12 +65,12 @@ public function test_on_plugin_update_message() {
 		$result = ob_get_clean();
 
 		// Assert
-		$this->assertRegExp( '/old version plugin/', $result );
-		$this->assertRegExp( '/invalid version plugin/', $result );
+		$this-> assertMatchesRegularExpression( '/old version plugin/', $result );
+		$this-> assertMatchesRegularExpression( '/invalid version plugin/', $result );
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/compatibility-tag/test-module.php:69]:
```suggestion
		$this->assertMatchesRegularExpression( '/old version plugin/', $result );
		$this->assertMatchesRegularExpression( '/invalid version plugin/', $result );
```

File: tests/phpunit/elementor/modules/compatibility-tag/test-module.php [hunk 30]
```diff
@@ -65,12 +65,12 @@ public function test_on_plugin_update_message() {
...
 		$this->assertNotRegExp( '/patch version plugin/', $result );
 		$this->assertNotRegExp( '/tested version plugin/', $result );
 		$this->assertNotRegExp( '/header not exists plugin/', $result );
-		$this->assertRegExp( '/extends elementor/', $result );
+		$this-> assertMatchesRegularExpression( '/extends elementor/', $result );
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/compatibility-tag/test-module.php:73]:
```suggestion
		$this->assertMatchesRegularExpression( '/extends elementor/', $result );
```

File: tests/phpunit/elementor/modules/dev-tools/test-deprecation.php [hunk 31]
```diff
@@ -10,7 +10,7 @@ class Test_Deprecation extends Elementor_Test_Base {
 	 */
 	private $deprecation;
 
-	public function setUp() {
+	public function setUp(): void  {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/dev-tools/test-deprecation.php:13]:
```suggestion
	public function setUp():void  {
```

File: tests/phpunit/elementor/modules/image-loading-optimization/test-module.php [hunk 32]
```diff
@@ -56,7 +56,7 @@ public function setUp() {
 	public function test_attachment_fixtures() {
 		$image = wp_get_attachment_image( self::$attachments[0], 'full' );
 
-		$this->assertRegExp( '/mock-image.png/', $image, 'Could not confirm attachments were created' );
+		$this-> assertMatchesRegularExpression( '/mock-image.png/', $image, 'Could not confirm attachments were created' );
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/image-loading-optimization/test-module.php:59]:
```suggestion
		$this->assertMatchesRegularExpression( '/mock-image.png/', $image, 'Could not confirm attachments were created' );
```

File: tests/phpunit/elementor/modules/kit-elements-defaults/import-export/runners/test-export.php [hunk 33]
```diff
@@ -19,7 +19,7 @@ public function setUp() {
 		parent::setUp();
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/kit-elements-defaults/import-export/runners/test-export.php:22]:
```suggestion
	public function tearDown():void  {
```

File: tests/phpunit/elementor/modules/library/documents/test-page.php [hunk 34]
```diff
@@ -9,7 +9,7 @@ class Elementor_Test_Page extends Elementor_Test_Base {
 	/** @var Page */
 	private static $page;
 
-	public function setUp() {
+	public function setUp(): void  {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/library/documents/test-page.php:12]:
```suggestion
	public function setUp():void  {
```

File: tests/phpunit/elementor/modules/library/documents/test-section.php [hunk 35]
```diff
@@ -9,7 +9,7 @@ class Elementor_Test_Section extends Elementor_Test_Base {
 	/** @var Section */
 	private static $section;
 
-	public static function setUpBeforeClass() {
+	public static function  setUpBeforeClass(): void {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/library/documents/test-section.php:12]:
```suggestion
	public static function setUpBeforeClass():void {
```

File: tests/phpunit/elementor/modules/site-navigation/data/recent/test-controller.php [hunk 36]
```diff
@@ -39,7 +39,7 @@ public function setUp() {
 
 	}
 
-	public function tearDown() {
+	public function  tearDown():void  {
```
Comment (TzviRabinovitch) [inline tests/phpunit/elementor/modules/site-navigation/data/recent/test-controller.php:42]:
```suggestion
	public function tearDown():void  {
```

**Review by louiswol94 — APPROVED** [review]

**Review by davseve — APPROVED** [review]

**Pull request merged**